### PR TITLE
add scripts to build aws-sdk-cpp s3 & s3-crt

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：

To add `aws-sdk-cpp` into thirdparty.  https://github.com/aws/aws-sdk-cpp

### CODE REVIEW

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -734,6 +734,15 @@ build_aliyun_oss_jars() {
     cp -r $TP_SOURCE_DIR/$ALIYUN_OSS_JARS_SOURCE $TP_INSTALL_DIR/aliyun_oss_jars
 }
 
+build_aws_cpp_sdk() {
+    check_if_source_exist $AWS_SDK_CPP_SOURCE
+    cd $TP_SOURCE_DIR/$AWS_SDK_CPP_SOURCE
+    # only build s3 and s3-crt, you can add more components if you want.
+    $CMAKE_CMD -Bbuild -DBUILD_ONLY="core;s3;s3-crt" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=${TP_INSTALL_DIR} -DENABLE_TESTING=OFF
```
Comment (imay) [inline thirdparty/build-thirdparty.sh:741]: What's the difference between s3 and s3-crt?

Comment (dirtysalt) [reply thirdparty/build-thirdparty.sh:741]: s3 is a standard s3 client, and s3-crt is the optimized version. 

https://github.com/aws/aws-sdk-cpp/wiki/Improving-S3-Throughput-with-AWS-SDK-for-CPP-v1.9

image

**Review by imay — DISMISSED** [review]
LGTM

**Comment by starrocks-xueli** [comment]
maybe you should add a license?

File: thirdparty/build-thirdparty.sh (hunk 1 above)
Comment (starrocks-xueli) [inline thirdparty/build-thirdparty.sh:741]: please add `transfer` to DBUILD_ONLY

Comment (starrocks-xueli) [reply thirdparty/build-thirdparty.sh:741]: please note that s3-crt is not stable according to my test.

Comment (dirtysalt) [reply thirdparty/build-thirdparty.sh:741]: OK.

**Review by imay — APPROVED** [review]
LGTM

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**